pkg/util/labels.go

```diff
@@ -60,27 +60,50 @@ func addDeploymentConfigNestedLabels(obj *deployapi.DeploymentConfig, labels lab
 	return nil
 }
 
+// addUnstructuredNestedLabels adds new label(s) to a nested labels of a single unstructured object
+func addUnstructuredNestedLabels(obj *runtime.Unstructured, labels labels.Set) error {
+	if template, ok := obj.Object["template"]; ok {
+		if m, ok := template.(map[string]interface{}); ok {
+			if controllerTemplate, ok := m["controllerTemplate"]; ok {
+				if m, ok := controllerTemplate.(map[string]interface{}); ok {
+					if podTemplate, ok := m["podTemplate"]; ok {
+						if m, ok := podTemplate.(map[string]interface{}); ok {
+							existing := make(map[string]string)
+							if l, ok := m["labels"]; ok {
+								if found, ok := extractLabels(l); ok {
+									existing = found
+								}
+							}
+							if err := MergeInto(existing, labels, OverwriteExistingDstKey); err != nil {
+								return err
+							}
+							m["labels"] = mapToGeneric(existing)
+						}
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
 // AddObjectLabels adds new label(s) to a single runtime.Object
 func AddObjectLabels(obj runtime.Object, labels labels.Set) error {
 	if labels == nil {
-		// Nothing to add
 		return nil
 	}
 
-	accessor, err := kmeta.Accessor(obj)
-	if err != nil {
-		return err
-	}
-
-	metaLabels := accessor.Labels()
-	if metaLabels == nil {
-		metaLabels = make(map[string]string)
-	}
+	if accessor, err := kmeta.Accessor(obj); err == nil {
```

Comment:
Fixed. `kmeta.Accessor` does not handle unstructured objects, so that's the scenario where we need to continue instead of returning the error right away.